.pre-commit-config.yaml

```diff
@@ -1,12 +1,69 @@
+# Learn more about this config here: https://pre-commit.com/
+
+# To enable these pre-commit hooks run:
+# `brew install pre-commit` or `python3 -m pip install pre-commit`
+# Then in the project root directory run `pre-commit install`
+
 default_stages: [commit]
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-builtin-literals
+      - id: check-case-conflict
+      - id: check-docstring-first
+      # - id: check-executables-have-shebangs  # TODO: Enable and fix or delete
+      # - id: check-json  # TODO: Enable and fix or delete
+      - id: check-shebang-scripts-are-executable
```

Comment:
This one doesn't seem too relevant, as I don't anticipate too many executable scripts.

Reply:
It is always the things that we do not anticipate that bite us... ;-)